File: LeetCode/DataStructure/Python/217.py
#https://leetcode.com/problems/contains-duplicate/

#__Simple Solution Using set
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        pre_l = len(nums)
        set_l = len(set(nums))
        if pre_l > set_l:
            return True
        else:
            return False

#__Hashing Method
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        item_count = {}
        for i in nums:
            if i in item_count:
                return True
            else:
                item_count[i] = 1
        return False

File: LeetCode/Study_Plan/DataStructure/C_Plus_Plus/217.cpp
//https://leetcode.com/problems/contains-duplicate/

class Solution
{
    public:
        bool containsDuplicate(vector<int>& nums)
        {
            unordered_map<int, int> hash_map;
            for (int i = 0; i < nums.size(); i++)
            {
                if( hash_map.find(nums[i]) == hash_map.end() )
                {
                    hash_map[nums[i]] = 1;
                }
                else
                {
                    return true;
                }
            }
            return false;

        }
};